tools/src/main/scala/scala/scalanative/build/LLVM.scala

```diff
@@ -85,8 +85,14 @@ private[scalanative] object LLVM {
       multithreadingEnabled ++ allowTargetOverrrides
     }
     val exceptionsHandling = {
-      val opt = if (isCpp) List("-fcxx-exceptions") else Nil
-      List("-fexceptions", "-funwind-tables") ::: opt
+      val targetSpecific = if (config.targetsWindows) {
+        val opt = if (isCpp) List("-fcxx-exceptions") else Nil
+        List("-fexceptions", "-funwind-tables") ++ opt
+      } else {
+        if (isCpp) List("-fno-rtti", "-fno-exceptions", "-funwind-tables")
```

Comment:
Should this be applied only to libunwind (or `nativelib`) code? Otherwise it is impossible to use c++ exceptions in c++ code in `scala-native` resources.

Reply:
Now that there is the config boolean to check if we are using the cpp runtime, we can use it here to selectively enable exceptions or not. To be discussed what flags we want.

Reply:
Probably the easiest thing to do is to change the if to `if (config.isCPPRuntimeEnabled)` so we enable exceptions to all cpp code if the C++ runtime is enabled.

Reply:
I think the two settings together make 4 options but some of the options don't make since with each other. e.g. if you take CPPExceptions then CPPRuntime needs to be true as well. Maybe they need to be a setting pair with some check of validity?

I also think that if we are not making a CPPRuntime then we should disallow compiling C++ in `scala-native` resources as because then we must link with C++.

It is unclear to me if a C++ lib has a C interface as with something like the C API for `libtensorflow` then would the runtime be C but then has a dependency on a C++ lib. Also, what if we try and link that statically. You might be able to test the default dynamic using this just to make sure - <https://github.com/ekrich/stensorflow.g8>